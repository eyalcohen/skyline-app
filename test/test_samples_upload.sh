#!/bin/sh -v -e

# unzip samples-db.zip -d /tmp
# mongod --dbpath /tmp/r0
# app.js
# ../util/delete_vehicle_data.js --vehicleId=1233922067

../util/samples_encode.js < samples.json > /tmp/samples.pbraw
curl -T /tmp/samples.pbraw -H 'Content-type: application/octet-stream' http://localhost:8080/samples
../util/query.js --vehicleId=1233922067 --json
