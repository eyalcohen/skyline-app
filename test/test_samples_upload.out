{
  "_schema": [
    {"beg":1322696276154000,"end":1322696276554000,"val":{"channelName":"vc/vcin8","humanName":"Id=0x19df7940,Req,Ch=3,DevId=VC,Param=3018,Op=getShort  (vc/vcin8)","type":"bool","merge":true}},
    {"beg":1322696276254000,"end":1322696276654000,"val":{"channelName":"mc/3001","humanName":"Id=0x11997722,Rsp,Ch=2,DevId=MC,Param=3001,Op=announceShort  (mc/3001)","type":"bytes","merge":false}},
    {"beg":1322696276354000,"end":1322696276754000,"val":{"channelName":"mc/hardFaults","humanName":"Id=0x11997722,Rsp,Ch=2,DevId=MC,Param=3001,Op=announceShort  (mc/hardFaults)","type":"bitfield","bitfieldBits":{"0":"faultUser","1":"faultOem","2":"faultSupplyUnderVoltage","3":"faultSupplyOverVoltage","4":"faultDriverUnderVoltage"},"merge":true}},
    {"beg":1322696276454000,"end":1322696276854000,"val":{"channelName":"pm/systemState","humanName":"Id=0x088afa60,Req,Ch=1,DevId=PM,Param=2003,Op=getShort  (pm/systemState)","type":"enum","enumVals":{"0":"KEY_OFF","1":"INITIALIZING","2":"ACTIVE","-1":"FAULTED"},"merge":true}},
    {"beg":1322696276554000,"end":1322696276954000,"val":{"channelName":"gps.speed_m_s","humanName":"GPS Speed","units":"m/s","type":"float","merge":false}},
  ],
  "gps.speed_m_s": [
    {"beg":1322696276554000,"end":1322696276654000,"val":1.2300000190734863},
    {"beg":1322696276654000,"end":1322696276754000,"val":2.3399999141693115},
    {"beg":1322696276754000,"end":1322696276854000,"val":3.450000047683716},
    {"beg":1322696276854000,"end":1322696276954000,"val":4.559999942779541},
  ],
  "mc/hardFaults": [
    {"beg":1322696276354000,"end":1322696276554000,"val":0},
    {"beg":1322696276554000,"end":1322696276654000,"val":2},
    {"beg":1322696276654000,"end":1322696276754000,"val":10},
  ],
  "pm/systemState": [
    {"beg":1322696276454000,"end":1322696276554000,"val":0},
    {"beg":1322696276554000,"end":1322696276754000,"val":2},
    {"beg":1322696276754000,"end":1322696276854000,"val":-1},
  ],
  "vc/vcin8": [
    {"beg":1322696276154000,"end":1322696276354000,"val":false},
    {"beg":1322696276354000,"end":1322696276554000,"val":true},
  ],
  "mc/3001": [
    {"beg":1322696276254000,"end":1322696276354000,"val":[3,2,1,0]},
    {"beg":1322696276354000,"end":1322696276454000,"val":[3,2,1,0]},
    {"beg":1322696276454000,"end":1322696276554000,"val":[4,0,0,0]},
    {"beg":1322696276554000,"end":1322696276654000,"val":[5,0,5,0]},
  ],
}
